src/test/java/ch/uzh/ifi/hase/soprafs22/lobby/LobbyTest.java

```diff
@@ -98,5 +101,37 @@ void updateLobbyPlayer_userNameNotUnique_conflict() throws FullLobbyException {
         assertThrows(DuplicateUserNameInLobbyException.class, () -> lobby.setUserName(host.getToken(), newPlayer.getName()));
     }
 
+    @Test
+    void generateUniqueName() throws LobbyNameConflictException, FullLobbyException {
+        // Create new lobby
+        Lobby lobby = new Lobby(0L, "lobbyName", Visibility.PRIVATE);
+        lobby.setGameMode(GameMode.TWO_VS_TWO);
+
+        lobby.getHost().setName("Player-0");
+        System.out.println(lobby.getHost().getName());
+
+        IPlayer player = lobby.generatePlayer();
+        player.setName("Player-1");
+        System.out.println(player.getName());
+        lobby.addPlayer(player);
+
+        player = lobby.generatePlayer();
+        player.setName("Player-2");
+        System.out.println(player.getName());
+        lobby.addPlayer(player);
+
+        player = lobby.generatePlayer();
+        player.setName("Player-2");
+        System.out.println(player.getName());
+        lobby.addPlayer(player);
+
+        System.out.println("");
+
+        for (IPlayer p : lobby) {
```

Comment:
I think you forgot the assert part of the test.

Reply:
Ahh, that was just for testing. I will remove that

Reply:
I quickly added two tests for testing unique name generation and resolving name conflict